src/Connection.php

```diff
@@ -912,15 +888,7 @@ public function fetchFirstColumn(string $query, array $params = [], array $types
      */
     public function iterateNumeric(string $query, array $params = [], array $types = []): Traversable
     {
-        try {
-            $result = $this->executeQuery($query, $params, $types);
-
-            while (($row = $result->fetchNumeric()) !== false) {
```

Comment:
We can call `iterateNumeric()` on the result instead of implementing iteration again.